### ISSUE TITLE
Qiskit installation fails with Python 3.9

### ISSUE THREAD

**Issue by EDDahl** [Qiskit/qiskit-metapackage#1111]
I followed the generic installation instructions here:

    https://qiskit.org/documentation/install.html

and installation failed on this step:

    pip install qiskit

There were thousands of lines of errors, which I will not attempt to include here.

To fix the problem, when building the base anaconda environment, it is necessary to specify:

    conda create -n name_of_my_env python=3.8

rather than what is specified in the installation instructions, which is:

    conda create -n name_of_my_env python=3

Apparently there is some incompatibility with the current version of qiskit & python 3.9.

### Informations

- **Qiskit version**: 0.23.0
- **Python version**: 3.9.0
- **Operating system**: macOS Catalina 10.15.7

### What is the current behavior?

Failure on this step:

    pip install qiskit

### Steps to reproduce the problem

    Follow steps in the generic installation instructions at URL given above.

### What is the expected behavior?

   A successful creation of a qiskit environment for development.

### Suggested solutions

  Specify python 3.8 rather than 3 which building the base conda environment.

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Details:  When issuing "pip install qiskit" here is the first error I encountered (note that there
were a total of 2734 lines of error output associated with this one error, so I'm only including
a few dozen lines here):

  ERROR: Command errored out with exit status 1:
   command: /Users/ddahl/anaconda3/envs/qiskit_new/bin/python /Users/ddahl/anaconda3/envs/qiskit_new/lib/python3.9/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/01/n14gcb491f563r3hzwx76y300000gn/T/pip-build-env-7pi9nc58/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'Cython>=0.28.5' 'numpy==1.13.3; python_version=='"'"'3.6'"'"' and platform_system!='"'"'AIX'"'"' and platform_python_implementation == '"'"'CPython'"'"'' 'numpy==1.14.0; python_version=='"'"'3.6'"'"' and platform_system!='"'"'AIX'"'"' and platform_python_implementation != '"'"'CPython'"'"'' 'numpy==1.14.5; python_version=='"'"'3.7'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.17.3; python_version>='"'"'3.8'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.6'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.7'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.17.3; python_version>='"'"'3.8'"'"' and platform_system=='"'"'AIX'"'"'' 'scipy>=0.19.1'
       cwd: None
  Complete output (2734 lines):
  Ignoring numpy: markers 'python_version == "3.6" and platform_system != "AIX" and platform_python_implementation == "CPython"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.6" and platform_system != "AIX" and platform_python_implementation != "CPython"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.7" and platform_system != "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.6" and platform_system == "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.7" and platform_system == "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version >= "3.8" and platform_system == "AIX"' don't match your environment
  Collecting setuptools
    Using cached setuptools-50.3.2-py3-none-any.whl (785 kB)
  Collecting wheel
    Using cached wheel-0.35.1-py2.py3-none-any.whl (33 kB)
  Collecting Cython>=0.28.5
    Using cached Cython-0.29.21-py2.py3-none-any.whl (974 kB)
  Collecting numpy==1.17.3
    Using cached numpy-1.17.3.zip (6.4 MB)
  Collecting scipy>=0.19.1
    Using cached scipy-1.5.4-cp39-cp39-macosx_10_9_x86_64.whl (29.1 MB)
  Building wheels for collected packages: numpy
    Building wheel for numpy (setup.py): started
    Building wheel for numpy (setup.py): finished with status 'error'

**Comment by corneliusroemer** [comment]
This issue lacks details on what exactly fails. In my case (Python 3.9.0 on WSL Ubuntu 20.04) it was the following error message, presumably because for 3.9 there's no qiskit-aer build available, and building from source fails:
```
    ERROR: Command errored out with exit status 1:
     command: /home/roemerco/miniconda3/envs/qiskit/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ggwv8x03/qiskit-aer/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ggwv8x03/qiskit-aer/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-exjf6aj2/install-record.txt --single-version-externally-managed --compile --install-headers /home/roemerco/miniconda3/envs/qiskit/include/python3.9/qiskit-aer
         cwd: /tmp/pip-install-ggwv8x03/qiskit-aer/
    Complete output (198 lines):


    --------------------------------------------------------------------------------
    -- Trying "Ninja" generator
    --------------------------------
    ---------------------------
    ----------------------
    -----------------
    ------------
    -------
    --
    Not searching for unused variables given on the command line.
    CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
    -- Configuring incomplete, errors occurred!
    See also "/tmp/pip-install-ggwv8x03/qiskit-aer/_cmake_test_compile/build/CMakeFiles/CMakeOutput.log".
    --
    -------
    ------------
    -----------------
    ----------------------
    ---------------------------
    --------------------------------
    -- Trying "Ninja" generator - failure
    --------------------------------------------------------------------------------



    --------------------------------------------------------------------------------
    -- Trying "Unix Makefiles" generator
    --------------------------------
    ---------------------------
    ----------------------
    -----------------
    ------------
    -------
    --
    Not searching for unused variables given on the command line.
    -- The C compiler identification is GNU 9.3.0
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Check for working C compiler: /usr/bin/cc - skipped
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- The CXX compiler identification is GNU 9.3.0
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Check for working CXX compiler: /usr/bin/c++ - skipped
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /tmp/pip-install-ggwv8x03/qiskit-aer/_cmake_test_compile/build
    --
    -------
    ------------
    -----------------
    ----------------------
    ---------------------------
    --------------------------------
    -- Trying "Unix Makefiles" generator - success
    --------------------------------------------------------------------------------

    Configuring Project
      Working directory:
        /tmp/pip-install-ggwv8x03/qiskit-aer/_skbuild/linux-x86_64-3.9/cmake-build
      Command:
        /tmp/pip-install-ggwv8x03/qiskit-aer/.eggs/cmake-3.18.4.post1-py3.9-linux-x86_64.egg/cmake/data/bin/cmake /tmp/pip-install-ggwv8x03/qiskit-aer -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-ggwv8x03/qiskit-aer/_skbuild/linux-x86_64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/home/roemerco/miniconda3/envs/qiskit/bin/python -DPYTHON_VERSION_STRING:STRING=3.9.0 -DPYTHON_INCLUDE_DIR:PATH=/home/roemerco/miniconda3/envs/qiskit/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/home/roemerco/miniconda3/envs/qiskit/lib/libpython3.9.so -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/home/roemerco/miniconda3/envs/qiskit/lib/python3.9/site-packages/skbuild/resources/cmake -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.9 -DCMAKE_BUILD_TYPE:STRING=Release

    -- The CXX compiler identification is GNU 9.3.0
    -- The C compiler identification is GNU 9.3.0
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Check for working CXX compiler: /usr/bin/c++ - skipped
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Check for working C compiler: /usr/bin/cc - skipped
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Performing Test flag_supported
    -- Performing Test flag_supported - Success
    -- Conan: Installing config from /tmp/pip-install-ggwv8x03/qiskit-aer/conan_settings
    ERROR: Unable to deduce type config install: /tmp/pip-install-ggwv8x03/qiskit-aer/conan_settings
    -- Conan: Automatic detection of conan settings from cmake
    -- Conan: Settings= -s;build_type=Release;-s;compiler=gcc;-s;compiler.version=9;-s;compiler.libcxx=libstdc++11
    -- Conan: checking conan executable
    -- Conan: Found program /home/roemerco/miniconda3/envs/qiskit/bin/conan
    -- Conan: Version found Conan version 1.31.2

    -- Conan executing: /home/roemerco/miniconda3/envs/qiskit/bin/conan install . -s build_type=Release -s compiler=gcc -s compiler.version=9 -s compiler.libcxx=libstdc++11 -e=CONAN_CMAKE_PROGRAM=/tmp/pip-install-ggwv8x03/qiskit-aer/.eggs/cmake-3.18.4.post1-py3.9-linux-x86_64.egg/cmake/data/bin/cmake -g=cmake --build=missing
    Configuration:
    [settings]
    arch=x86_64
    arch_build=x86_64
    build_type=Release
    compiler=gcc
    compiler.libcxx=libstdc++11
    compiler.version=9
    os=Linux
    os_build=Linux
    [options]
    [build_requires]
    [env]
    CONAN_CMAKE_PROGRAM=/tmp/pip-install-ggwv8x03/qiskit-aer/.eggs/cmake-3.18.4.post1-py3.9-linux-x86_64.egg/cmake/data/bin/cmake
    conanfile.txt: Installing package
    Requirements
        fmt/6.2.1 from 'conan-center' - Cache
        muparserx/4.0.8 from 'conan-center' - Cache
        nlohmann_json/3.1.1 from 'conan-center' - Cache
        spdlog/1.5.0 from 'conan-center' - Cache
    Packages
        fmt/6.2.1:b911f48570f9bb2902d9e83b2b9ebf9d376c8c56 - Cache
        muparserx/4.0.8:b911f48570f9bb2902d9e83b2b9ebf9d376c8c56 - Cache
        nlohmann_json/3.1.1:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
        spdlog/1.5.0:942d5c94aa934511ee4500bda27908cb4e791b24 - Cache

    Installing (downloading, building) binaries...
    fmt/6.2.1: Already installed!
    muparserx/4.0.8: Already installed!
    nlohmann_json/3.1.1: Already installed!
    spdlog/1.5.0: Already installed!
    conanfile.txt: Generator cmake created conanbuildinfo.cmake
    conanfile.txt: Generator txt created conanbuildinfo.txt
    conanfile.txt: Generated conaninfo.txt
    conanfile.txt: Generated graphinfo
    -- Conan: Loading conanbuildinfo.cmake
    -- Conan: Adjusting output directories
    -- Conan: Using cmake targets configuration
    -- Library spdlog found /home/roemerco/.conan/data/spdlog/1.5.0/_/_/package/942d5c94aa934511ee4500bda27908cb4e791b24/lib/libspdlog.a
    -- Library muparserx found /home/roemerco/.conan/data/muparserx/4.0.8/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/lib/libmuparserx.a
    -- Library fmt found /home/roemerco/.conan/data/fmt/6.2.1/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/lib/libfmt.a
    -- Conan: Adjusting language standard
    -- Current conanbuildinfo.cmake directory: /tmp/pip-install-ggwv8x03/qiskit-aer/_skbuild/linux-x86_64-3.9/cmake-build
    -- Conan: Compiler GCC>=5, checking major version 9
    -- Conan: Checking correct version: 9
    -- Looking for OpenMP support...
    -- OpenMP found!
    -- OpenMP_CXX_FLAGS = -fopenmp
    -- OpenMP_EXE_LINKER_FLAGS =
    -- Looking for pthread.h
    -- Looking for pthread.h - found
    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
    -- Looking for pthread_create in pthreads
    -- Looking for pthread_create in pthreads - not found
    -- Looking for pthread_create in pthread
    -- Looking for pthread_create in pthread - found
    -- Found Threads: TRUE
    -- Looking for OpenBLAS library...
    -- OpenBLAS not found. Looking for any other BLAS & Lapack libraries...
    -- Looking for sgemm_
    -- Looking for sgemm_ - not found
    CMake Error at .eggs/cmake-3.18.4.post1-py3.9-linux-x86_64.egg/cmake/data/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
      Could NOT find BLAS (missing: BLAS_LIBRARIES)
    Call Stack (most recent call first):
      .eggs/cmake-3.18.4.post1-py3.9-linux-x86_64.egg/cmake/data/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
      .eggs/cmake-3.18.4.post1-py3.9-linux-x86_64.egg/cmake/data/share/cmake-3.18/Modules/FindBLAS.cmake:947 (find_package_handle_standard_args)
      CMakeLists.txt:209 (find_package)


    -- Configuring incomplete, errors occurred!
    See also "/tmp/pip-install-ggwv8x03/qiskit-aer/_skbuild/linux-x86_64-3.9/cmake-build/CMakeFiles/CMakeOutput.log".
    See also "/tmp/pip-install-ggwv8x03/qiskit-aer/_skbuild/linux-x86_64-3.9/cmake-build/CMakeFiles/CMakeError.log".
    Traceback (most recent call last):
      File "/home/roemerco/miniconda3/envs/qiskit/lib/python3.9/site-packages/skbuild/setuptools_wrap.py", line 582, in setup
        env = cmkr.configure(cmake_args,
      File "/home/roemerco/miniconda3/envs/qiskit/lib/python3.9/site-packages/skbuild/cmaker.py", line 229, in configure
        raise SKBuildError(

    An error occurred while configuring with CMake.
      Command:
        /tmp/pip-install-ggwv8x03/qiskit-aer/.eggs/cmake-3.18.4.post1-py3.9-linux-x86_64.egg/cmake/data/bin/cmake /tmp/pip-install-ggwv8x03/qiskit-aer -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-ggwv8x03/qiskit-aer/_skbuild/linux-x86_64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/home/roemerco/miniconda3/envs/qiskit/bin/python -DPYTHON_VERSION_STRING:STRING=3.9.0 -DPYTHON_INCLUDE_DIR:PATH=/home/roemerco/miniconda3/envs/qiskit/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/home/roemerco/miniconda3/envs/qiskit/lib/libpython3.9.so -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/home/roemerco/miniconda3/envs/qiskit/lib/python3.9/site-packages/skbuild/resources/cmake -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.9 -DCMAKE_BUILD_TYPE:STRING=Release
      Source directory:
        /tmp/pip-install-ggwv8x03/qiskit-aer
      Working directory:
        /tmp/pip-install-ggwv8x03/qiskit-aer/_skbuild/linux-x86_64-3.9/cmake-build
    Please see CMake's output for more information.
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/roemerco/miniconda3/envs/qiskit/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ggwv8x03/qiskit-aer/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ggwv8x03/qiskit-aer/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-exjf6aj2/install-record.txt --single-version-externally-managed --compile --install-headers /home/roemerco/miniconda3/envs/qiskit/include/python3.9/qiskit-aer Check the logs for full command output.
```

Limited workaround: Downgrade Python to 3.8, then the install should work.

**Comment by insominiac21** [comment]
I guess qiskit is yet to support 3.9, try with 3.8 or something

**Comment by mtreinish** [comment]
Since qiskit 0.23.3 Python 3.9 has been supported. That was the first release to include precompiled wheels for qiskit, prior to that we were blocked waiting on upstream dependencies to support python 3.9 first. I am going to close this issue now as 3.9 support has been presetin a released version for ~20 days now. If you're still having issues installing please open a new issue with the details
